### PR TITLE
Reform to remove head of household filing status

### LINKED ISSUE #3162: Reform to remove head of household filing status
https://taxfoundation.org/research/all/federal/growth-opportunity-us-tax-reform-plan/#Revenue

### PR DESCRIPTION
**Pull request by mzx-4real**
Fixes #3162

Co-authored-by: Pavel Makarchuk <PavelMakarchuk@users.noreply.github.com>

### Summary
:sparkles::page_facing_up::hammer:

This pull request adds a new reform option for the Tax Foundation growth and opportunity plan, which removes the head of household filing status and adjusts the standard deduction accordingly. It creates a new parameter file, a new reform package, and a new reform module that define and implement the reform logic. It also updates the changelog file with a new entry.

> _New tax reform plan_
> _`growth_and_opportunity`_
> _Autumn of changes_

### Walkthrough
*  Add a new reform for the Tax Foundation growth and opportunity plan ([link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R5), [link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-d6d0964d31d3e40b921dc8ddada69b08b9ab435a935bc274dc0377e3a50144aaR1-R9), [link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-4e9d02babed545dde6927fb44143a91cb570bb3a05ff35045f0d2b887de25804R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-82018c2c641e3799661509f050697f56fe68599d181558ccdf9180547a34bb1aR1-R132))
  - Create a new parameter file `remove_head_of_household.yaml` with the option to remove the head of household filing status ([link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-d6d0964d31d3e40b921dc8ddada69b08b9ab435a935bc274dc0377e3a50144aaR1-R9))
  - Create a new Python package `growth_and_opportunity` with an empty `__init__.py` file ([link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-4e9d02babed545dde6927fb44143a91cb570bb3a05ff35045f0d2b887de25804R1))
  - Create a new reform module `remove_standard_deduction_head_of_household.py` that overrides the standard deduction variables and updates them based on the parameter value and the American Family Act reform ([link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-82018c2c641e3799661509f050697f56fe68599d181558ccdf9180547a34bb1aR1-R132))
  - Create a reform object and a test instance of the reform in the reform module ([link](https://github.com/PolicyEngine/policyengine-us/pull/3163/files?diff=unified&w=0#diff-82018c2c641e3799661509f050697f56fe68599d181558ccdf9180547a34bb1aR1-R132))

### CODE REVIEW

**Review by MaxGhenis — COMMENTED** [review]
is this only for the standard deduction? hoh also has different brackets. you can address both pieces by changing the formula for `filing_status` instead

File: policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py
Comment (PavelMakarchuk) [inline policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:1]: now we need to add this reform to a list of reforms in the `reforms.py` file and add a unit test to the remove_standard_deduction_head_of_household variable

File: policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    filing_status == statuses.JOINT,
+                    filing_status == statuses.WIDOW,
+                    filing_status == statuses.HEAD_OF_HOUSEHOLD,
+                ],
+                [
+                    std.amount[filing_status],
+                    std.amount[filing_status],
+                    std.amount[filing_status],
+                    std.amount[filing_status],
+                    13850,  # this should be single amount, use fix amount as temp
```
Comment (PavelMakarchuk) [inline policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:40]: @MaxGhenis how can we extract the single value from a breakdown here? Any suggestions for a clean solution without creating 5 separate files?

Comment (PavelMakarchuk) [reply policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:40]: @MaxGhenis friendly reminder

File: policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    def formula(tax_unit, period, parameters):
+        std = parameters(period).gov.irs.deductions.standard
+        filing_status = tax_unit("filing_status", period)
+        statuses = filing_status.possible_values
+        separate_filer_itemizes = tax_unit("separate_filer_itemizes", period)
+        claimed_as_dependent_elsewhere = tax_unit(
+            "tax_unit_dependent_elsewhere", period
+        )
+        if (
```
Comment (PavelMakarchuk) [inline policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:21]: get rid of the if statement and rewrite this variable as if it is true

File: policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        unit = USD
+        documentation = "The additional standard deduction under the tax foundation growth and opportunity plan"
+        definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        std = parameters(period).gov.irs.deductions.standard
+        filing_status = tax_unit("filing_status", period)
+        statuses = filing_status.possible_values
+        aged_blind_count = tax_unit("aged_blind_count", period)
+        if (
```
Comment (PavelMakarchuk) [inline policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:81]: same thing here

File: policyengine_us/tests/policy/contrib/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+- name: Remove standard deduction head of household filing status
+  period: 2023
+  reforms: policyengine_us.reforms.tax_foundation.growth_and_opportunity.remove_standard_deduction_head_of_household.remove_standard_deduction_head_of_household
+  input:
+    gov.contrib.tax_foundation.growth_and_opportunity.remove_head_of_household : True
+    # tax unit basic standard deduction
+    filing_status: HEAD_OF_HOUSEHOLD
+    separate_filer_itemizes: False
+    tax_unit_dependent_elsewhere: False
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/contrib/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.yaml:9]:
```suggestion
    tax_unit_dependent_elsewhere: false
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
please adjust the formula for filing_status instead: https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/household/demographic/tax_unit/filing_status.py

Don't need to touch standard deduction. The TF report discusses removing head of household filing status entirely, which also includes brackets etc.

File: policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        entity = TaxUnit
+        possible_values = FilingStatus
+        default_value = FilingStatus.SINGLE
+        definition_period = YEAR
+        label = "Filing_status (Eliminating HoH)"
+        documentation = "Eliminating HoH under the tax foundation growth and opportunity plan"
+
+        def formula(tax_unit, period, parameters):
+            has_spouse = add(tax_unit, period, ["is_tax_unit_spouse"]) > 0
+            has_dependents = tax_unit("tax_unit_dependents", period) > 0
```
Comment (MaxGhenis) [inline policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:23]: We don't need dependents if no hoh

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Can you also try to find who first proposed hoh repeal? Many have done so, including Senator Romney in his Family Security Act

**Comment by mzx-4real** [comment]
> Can you also try to find who first proposed hoh repeal? Many have done so, including Senator Romney in his Family Security Act

@MaxGhenis  Hi max. I think for this particular reform of repealing hoh and changing standard deduction, there isn't an exact senator that proposed this. Rather it is proposed by tax foundation's analysts. The link of the reform analysis is [here](https://taxfoundation.org/research/all/federal/growth-opportunity-us-tax-reform-plan/#Revenue). The projection is in Table 6 (Expand the standard deduction to $19,500 (double for joint filers) starting in 2024 and eliminate Head of Household filing status).

File: policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        label = "Filing_status (Eliminating HoH)"
+        documentation = "Eliminating HoH under the tax foundation growth and opportunity plan"
+
+        def formula(tax_unit, period, parameters):
+            has_spouse = add(tax_unit, period, ["is_tax_unit_spouse"]) > 0
+            person = tax_unit.members
+            is_separated = tax_unit.any(person("is_separated", period))
+            is_widowed = tax_unit.any(person("is_widowed", period))
+            return select(
+                [
```
Comment (MaxGhenis) [inline policyengine_us/reforms/tax_foundation/growth_and_opportunity/remove_standard_deduction_head_of_household.py:27]:
```suggestion
                [has_spouse,
                    is_separated,
                    is_widowed], [FilingStatus.JOINT,
                    FilingStatus.SEPARATE,
                    FilingStatus.WIDOW], default=FilingStatus.SINGLE
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
I understand Tax Foundation proposed repealing Head of Household filing status in June 2023. They were not the first to do so. For example, [Senator Romney did so in February 2021](https://www.romney.senate.gov/wp-content/uploads/2021/02/family-security-act_one-pager.pdf). Could you dig around to see if others proposed it previously? If we can't find others then let's attribute it to Romney.

**Comment by MaxGhenis** [comment]
Let's attribute this to Senator Romney's Family Security Act, so put in the `congress` folder

**Comment by MaxGhenis** [comment]
I'll finish this up as it's come up for a business need

**Pull request merged**